velox/connectors/hive/storage_adapters/s3fs/S3FileSystem.cpp

```diff
@@ -137,15 +138,24 @@ namespace filesystems {
 
 namespace S3Config {
 namespace {
+// S3 configuration parameters are used as keys into the config map.
 constexpr char const* kPathAccessStyle{"hive.s3.path-style-access"};
 constexpr char const* kEndpoint{"hive.s3.endpoint"};
 constexpr char const* kSecretKey{"hive.s3.aws-secret-key"};
 constexpr char const* kAccessKey{"hive.s3.aws-access-key"};
 constexpr char const* kSSLEnabled{"hive.s3.ssl.enabled"};
+constexpr char const* kS3IAMRole{"hive.s3.iam-role"};
+constexpr char const* kS3IAMRoleSessionName{"hive.s3.iam-role-session-name"};
 constexpr char const* kUseInstanceCredentials{
     "hive.s3.use-instance-credentials"};
 } // namespace
 } // namespace S3Config
+namespace S3ConfigDefaults {
```

Comment:
Do we need a separate namespace for this?

Reply:
I felt, instead of using the term `Default` for each of the variable names, we could use this separate namespace and use the same variable names. I am open to either approach here.
Example:
```
namespace S3Config {
namespace {
  constexpr char const* kS3IAMRoleSessionName{"hive.s3.iam-role-session-name"};
}
}
namespace S3ConfigDefaults {
namespace {
  const std::string kS3IAMRoleSessionName{"presto-session"};
}
}
```

Reply:
I guess I must mention that over time the list here will keep increasing. See S3 defaults on the Java side
https://github.com/prestodb/presto/blob/master/presto-hive/src/main/java/com/facebook/presto/hive/s3/HiveS3Config.java#L33

Reply:
I think using Default suffix in the names is better. The namespace names should be lowercase. It would be nice to change S3Config to s3config. Alternatively, I'd consider introducing an S3Config class with a getter per config property and hiding the property named inside, e.g.

```
   auto s3Config = S3Config(config_);
   bool b = s3Config.useInstanceCredentials();
   std::string rule = s3Config.iamRole();
   ...
```

Reply:
Good idea! I will implement the getter API. Do you want me to make that change in this PR or a follow-up one?

Reply:
It would be nice to introduce S3Config in a separate PR, then use it here.